my-app/src/context/user-context.js

```diff
@@ -3,14 +3,16 @@ import userReducer from "./reducers/userReducer";
 import addToCart from "./functions/addToCart";
 import updateQuantity from "./functions/updateQuantityInCart";
 import deleteProduct from "./functions/deleteFromCart";
+import addToWishlist from "./functions/addToWishlist";
+import removeFromWishlist from "./functions/removeFromWishlist";
 
 const UserContext = createContext()
 
 const UserContextProvider = ({children}) => {
     const [state,dispatch] = useReducer(userReducer,{encodedToken : "" , cart : [] , wishlist : [],showLoginAlert : false})
-    console.log("user context state",state)
```

Comment:
I liked how you have used the showLoginAlert state to handle unauthorized users.